src/mudlet-lua/tests/DB_spec.lua

```diff
@@ -371,18 +371,36 @@ describe("Tests DB.lua functions", function()
       assert.are.same(test, res[1])
     end)
 
-    it("should successfully delete columns in a non empty table.",
+    it("should successfully delete columns in a non empty table if force argument is provided.",
```

Comment:
Just to check... this is changing backwards compatibility to be an opt-in?

Reply:
Yes. I made this choice because I considered it safer to reject by default instead of potentially deleting wanted data; the argument also doubles as a way to override the check if the user is certain they want to drop the columns.

If backwards compatibility is more important, I can change the force argument to a check argument.

Reply:
To be fair....this is a case where what the documentation implies and the way the current code actually works is in direct conflict.

On https://wiki.mudlet.org/w/Manual:Database_Functions#db:create it says: "If the named database does not exist it will create it. If the database does exist then it will add any columns or indexes which didn’t exist before to that database. If the database already has all the specified columns and indexes, it will do nothing."

To me, that implied that it doesn't delete anything...but the current code will delete columns that doesn't match, easily resulting in lost data if you make a minor change in the db:create line.

I personally think any data loss in a database should be done purposefully/explicitly, or confirmed after a warning is issued.

Reply:
I'm in agreement, the players at the end of the day will be better off with this approach. @demonnic and @keneanung?

Reply:
I agree, I think this will be better in the end